doc/whats_new/v0.20.rst

```diff
@@ -1253,6 +1253,10 @@ Miscellaneous
   happens immediately (i.e., without a deprecation cycle).
   :issue:`11741` by `Olivier Grisel`_.
 
+- |Fix| Fixed a bug in validation.py where :func: `_num_samples()` would not properly
```

Comment:
We usually avoid referencing private functions here. How about just "a bug in validation helpers where passing a Dask DataFrame would result in an error"?